StartBonusActivationOrchestrator/handler.ts

```diff
@@ -153,6 +166,17 @@ export const getStartBonusActivationOrchestratorHandler = (
       }
     } catch (e) {
       context.log.error(`${logPrefix}|ID=${operationId}|ERROR=${toString(e)}`);
+
+      // release lock in case the orchestrator fails
+      yield context.df.callActivityWithRetry(
```

Comment:
if any Activity inside the try block fails, the lock will be removed. IMHO is an unwanted behaviour. Only if `SendBonusActivationActivity` fail the lock must be removed.

Reply:
also in case we are on the `FailedBonusActivation` branch

Reply:
yes

Reply:
fixed